### PR TITLE
Resolve issue 145 without .so files

### PR DESCRIPTION
**Pull request by dannytoomey**
Hi Witold,

Here's a new PR that resolves issue #145 without .so files in the commit history. 

Best regards,

Danny Toomey

### CODE REVIEW

**Comment by dannytoomey** [comment]
Don't merge yet, this does not pass build checks.

File: R/trt_effects.R [hunk 1]
```diff
@@ -110,3 +110,43 @@ mutau_cor <- function(bg,
...
+ #' @param bg        A [baggr] model.
+ #' @param transform A transformation to apply to the result, should be an R function;
+ #'                  (this is commonly used when calling `hypermean` from other
+ #'                  plotting or printing functions)
+ #' @param interval  The confidence inerval of the hypermean passed as any value between 0 and 1. 
+ #' @param message   A logical agrument that determines whether the result is printed or not.  
+ #' @return The hypermean of the `baggr` model with the interval specified by `interval`, with 
+ #'         a default interval of 95%, as calculated by [treatment_effect].
+ #' @export
+ hypermean <- function(bg,transform=NULL,interval = 0.95,message=TRUE) {
```
Comment (wwiecek) [inline R/trt_effects.R:124]: (1) why not do `message = FALSE` by default?
(2) I'd add argument `summary=TRUE` and then `treatment_effect(bg, summary=summary, ...)`; that way the default behaviour is the same as what you programmed, but in the future I can also call something like `hypersd(bg, s=F)`, seems handy

File: tests/testthat/test_binary.R [hunk 2]
```diff
@@ -138,8 +138,8 @@ test_that("extra pooling stats work", {
 test_that("Calculation of effects works", {
   expect_is(group_effects(bg5_p), "array")
   expect_is(treatment_effect(bg5_p), "list")
-  expect_length(treatment_effect(bg5_p, summary = TRUE)$tau, 5)
```
Comment (wwiecek) [inline tests/testthat/test_binary.R:141]: YOu don't want to overwrite your tests, you want to keep the old tests on `treatment_effect` and then write additional tests to check if `hypermean(bg)` is equal to `treatment_effect(bg, s=T)[[1]]` etc etc. You want to do it for several models/cases, just to be extra sure

**Review by wwiecek — CHANGES_REQUESTED** [review]

File: R/trt_effects.R [hunk 3]
```diff
@@ -110,3 +110,43 @@ mutau_cor <- function(bg,
 
 }
 
+#' Return the hypermean of the treatments effects
```
[the remark this answers is not in the document]
Comment (dannytoomey) [reply R/trt_effects.R:113]: Got it! I linked the docs for `hypermean` and `hypersd` to `treatment_effects`, so there's only one .Rd file

File: R/trt_effects.R (hunk 1 above)
Comment (dannytoomey) [reply R/trt_effects.R]: 1) No real reason, I switched to the be false by default 
2) Agreed, seems handy. Added this.

**Comment by wwiecek** [comment]
This has now been incorporated into `devel` without merging

**Pull request closed without merging**